packages/toolpad-app/src/toolpad/AppEditor/PagePanel.tsx

```diff
@@ -82,6 +84,20 @@ export interface ComponentPanelProps {
 export default function PagePanel({ appId, className, sx }: ComponentPanelProps) {
   const { data: app, isLoading } = client.useQuery('getApp', [appId]);
 
+  const [latestStoredApp, setLatestStoredApp] = useLocalStorageState<LatestStoredAppValue>(
+    TOOLPAD_LATEST_APP_KEY,
+    null,
+  );
+
+  React.useEffect(() => {
+    if (app && (!latestStoredApp || appId !== latestStoredApp.appId)) {
+      setLatestStoredApp({
```

Comment:
I wanted it to work also if you load an app, not just when you create it.

Reply:
But before loading it you would have to create it, no? If I understand correctly this would only benefit someone who hasn't yet closed the tab and received this update - like they do still have Toolpad open in their browser, otherwise if you come to Toolpad after closing the tab your app would be gone and you would have to go through creation process. If that's the only case then IMO it would be more straight forward and less complex to do that during creation

Reply:
The latest app isn't gone if you close the tab, it's saved in `localStorage`, so the last app that you were working on (not necessarily the last app you created) will always be there to go back to when you load the demo home page.

In the context of how the demo works right now there's no big difference, but I think that if we set this value when an app loads then we can always rely on it in the future to get the latest app the user was working on, for any future use case.
In my opinion there's no benefit in setting this when the user creates the app as opposed to when the app loads.

Reply:
I'm hesitant for this. Imagine having two tabs open with different apps, it  makes these two tabs fight with each other over who's the latest app. I'm also more inclined towards only storing the last created app here.

Reply:
> Imagine having two tabs open with different apps

Wouldn't that be an uncommon use case though, and the only negative thing that could happen is that a recent app would be suggested instead of really the latest one that was used?

Anyway I can move it to when the app is created, it should be virtually the same for now.